### PR TITLE
[WIP] Feature/adding postgres ssl connection option

### PR DESCRIPTION
**Pull request by darwinsubramaniam**
# Description
Enabling the SSL connection option for Postgres.

Fixes # (issue)
#868

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] I have tested locally
- [x] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [x] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch

### CODE REVIEW

File: packages/node-core/src/db/db.module.ts [hunk 1]
```diff
@@ -23,6 +32,42 @@ const DEFAULT_DB_OPTION: DbOption = {
...
+USER should provide the path as environment variable
+- DB_CA_CERT: path to the CA file (Required)
+- DB_KEY_CERT: path to the KEY file (Optional)
+- DB_CLIENT_CERT: path to the CERT file (Optional)
+@return the option with ssl option
+@exception throw error if the ssl is enabled but there is not correct file path
+*/
+const dbOptionWithSSLOptions = (option: DbOption): DbOption => {
+  if (option.ssl) {
+    const ca_content = readFileSync(process.env.DB_CA_CERT).toString();
```
Comment (stwiname) [inline packages/node-core/src/db/db.module.ts:48]: Can you please add some error handling for this.

File: packages/node-core/src/db/db.module.ts [hunk 2]
```diff
@@ -23,6 +32,42 @@ const DEFAULT_DB_OPTION: DbOption = {
   username: process.env.DB_USER ?? 'postgres',
   password: process.env.DB_PASS ?? 'postgres',
   database: process.env.DB_DATABASE ?? 'postgres',
+  ssl: process.env.DB_SSL === 'true' ?? false,
```
Comment (stwiname) [inline packages/node-core/src/db/db.module.ts:35]: Rather than having `DB_SSL`, we can use whether `DB_CA_CERT` is defined as the check to use ssl or not.

**Review by stwiname — CHANGES_REQUESTED** [review]
Hi @darwinsubramaniam 

Thanks for making this PR.

I think for the connection options it would be great to do it through the args, [and by extension it adds support for env args](https://github.com/subquery/subql/blob/main/packages/node/src/yargs.ts#L9).

It would also be self documenting with the help command and add some type checking.

File: packages/node-core/src/db/db.module.ts (hunk 1 above)
Comment (darwinsubramaniam) [reply packages/node-core/src/db/db.module.ts:48]: > 

Hi @stwiname ,
It is great suggestion, I will work on the modification as per advice and together with the error handling

Comment (darwinsubramaniam) [reply packages/node-core/src/db/db.module.ts]: introduced error handling by logging error, but the file reading content is common function for both  node-core and query package, so introduced the new function in common package and leveraged that for both node-core and query.

**Comment by darwinsubramaniam** [comment]
Hi @stwiname ,

Made changes as per suggested , using the args and the required error  handling by logging and exit the program. 

Pending Task: 

- [ ]  Test using the ssl Postgres.

File: packages/node-core/src/configure/NodeConfig.ts [hunk 3]
```diff
@@ -201,6 +204,50 @@ export class NodeConfig implements IConfig {
     return this._config.unfinalizedBlocks;
   }
 
+  get isPostgresSecureConnection(): boolean {
+    if (this._config.pgCa) {
+      return true;
+    }
+    return false;
```
Comment (stwiname) [inline packages/node-core/src/configure/NodeConfig.ts:211]:
```suggestion
    return !!this._config.pgCa;
```

**Review by stwiname — CHANGES_REQUESTED** [review]
Nice work!

We will do some testing on our end.

**Pull request merged**